From 3afebc7cfe95bf29a3f0acf33ff68a7c1cd9b3c5 Mon Sep 17 00:00:00 2001
From: Harman Kalra <hkalra@marvell.com>
Date: Mon, 20 May 2024 19:09:58 +0530
Subject: [PATCH 2/2] netdev-dpdk: valid lcore id for main thread

Overwriting the lcore ID in main thread may result in issue which main
thread participates in fast path

Signed-off-by: Harman Kalra <hkalra@marvell.com>
---
 lib/dpdk.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/lib/dpdk.c b/lib/dpdk.c
index d76d53f8f..f9eb4ec5a 100644
--- a/lib/dpdk.c
+++ b/lib/dpdk.c
@@ -437,9 +437,6 @@ dpdk_init__(const struct smap *ovs_other_config)
                              dpdk_unixctl_mem_stream,
                              malloc_dump_stats_wrapper);
 
-    /* We are called from the main thread here */
-    RTE_PER_LCORE(_lcore_id) = NON_PMD_CORE_ID;
-
     /* Finally, register the dpdk classes */
     netdev_dpdk_register(ovs_other_config);
     netdev_register_flow_api_provider(&netdev_offload_dpdk);
-- 
2.25.1

